packages/combo-box/src/vaadin-combo-box-mixin.js

```diff
@@ -1094,6 +1094,12 @@ export const ComboBoxMixin = (subclass) =>
 
     /** @private */
     _onFocusout(event) {
+      // VoiceOver on iOS fires `focusout` event when moving focus to the item in the dropdown.
+      // Do not focus the input in this case, because it would break announcement for the item.
+      if (event.relatedTarget && this._getItemElements().includes(event.relatedTarget)) {
```

Comment:
Changed to use `_getItemElements()` because we don't have `_tagNamePrefix` in `23.1` and older branches:
this getter is only present in 23.2 - it was added while removing `vaadin-combo-box-dropdown`.